src/expr/src/scalar/mod.rs

```diff
@@ -742,6 +742,53 @@ impl MirScalarExpr {
                         {
                             // Canonically order elements so that deduplication works better.
                             mem::swap(expr1, expr2);
+                        } else if let (
+                            BinaryFunc::Eq,
+                            MirScalarExpr::Literal(
+                                Ok(lit_row),
+                                ColumnType {
+                                    scalar_type:
+                                        ScalarType::Record {
+                                            fields: field_types,
+                                            ..
+                                        },
+                                    ..
+                                },
+                            ),
+                            MirScalarExpr::CallVariadic {
+                                func: VariadicFunc::RecordCreate { .. },
+                                exprs: rec_create_args,
+                            },
+                        ) = (func, (**expr1).clone(), (**expr2).clone())
+                        {
+                            // Literal(record(c1, c2)) = record_create(e1, e2)
+                            //  -->
+                            // c1 = e1 AND c2 = e2
+                            //
+                            // `literal_constraints` relies on this transform, because
```

Comment:
nit: qualify `literal_constraints` with the module or type on which it is defined.

Reply:
Done